src/lib/ilqp.js

```diff
@@ -0,0 +1,202 @@
+  ])
+}
+
+function _getQuote ({
+  plugin,
+  connector,
+  quoteQuery,
+  timeout
+}) {
+  const prefix = plugin.getInfo().prefix
+
+  debug('remote quote connector=' + connector, 'query=' + JSON.stringify(quoteQuery))
+  return _sendAndReceiveMessage({
+    plugin: plugin,
+    method: 'quote_response',
+    timeout: timeout,
+    message: {
+      ledger: prefix,
+      account: connector,
+      data: {
+        method: 'quote_request',
+        data: quoteQuery
+      }
+    }
+  }).then((response) => {
+    return response.data.data
+  })
+}
+
+function _getCheaperQuote (quote1, quote2) {
+  const source1 = new BigNumber(quote1.source_amount)
+  const dest1 = new BigNumber(quote1.destination_amount)
+
+  if (source1.lessThan(quote2.source_amount)) return quote1
+  if (dest1.greaterThan(quote2.destination_amount)) return quote1
+
+  return quote2
+}
+
+/**
+  * @param {Object} plugin The LedgerPlugin used to send quote request
+  * @param {Object} query
+  * @param {String} query.sourceAddress Sender's address
+  * @param {String} query.destinationAddress Recipient's address
+  * @param {String} [query.sourceAmount] Either the sourceAmount or destinationAmount must be specified
+  * @param {String} [query.destinationAmount] Either the sourceAmount or destinationAmount must be specified
+  * @param {String|Number} [query.sourceExpiryDuration] Number of seconds between when the source transfer is proposed and when it expires.
+  * @param {String|Number} [query.destinationExpiryDuration] Number of seconds between when the destination transfer is proposed and when it expires.
+  * @param {Array} [query.connectors] List of ILP addresses of connectors to use for this quote.
+  * @returns {Promise<Quote>}
+  */
+function * quote (plugin, {
+  sourceAddress,
+  destinationAddress,
+  sourceAmount,
+  destinationAmount,
+  sourceExpiryDuration,
+  destinationExpiryDuration,
+  connectors,
+  timeout
+}) {
+  if (!xor(sourceAmount, destinationAmount)) {
+    throw new Error('should provide source or destination amount but not both' +
+      ' ' + JSON.stringify({ sourceAmount, destinationAmount }))
+  }
+
+  yield safeConnect(plugin)
+  const prefix = plugin.getInfo().prefix
+  const amount = sourceAmount || destinationAmount
+
+  if (startsWith(prefix, destinationAddress)) {
+    debug('returning a local transfer to', destinationAddress, 'for', amount)
+    return omitUndefined({
+      sourceAmount: amount,
+      destinationAmount: amount,
+      sourceExpiryDuration: destinationExpiryDuration
+    })
+  }
+
+  const quoteQuery = omitUndefined({
+    source_address: plugin.getAccount(),
+    source_amount: sourceAmount,
+    destination_address: destinationAddress,
+    destination_amount: destinationAmount,
+    destination_expiry_duration: destinationExpiryDuration,
+  })
+
+  const quoteConnectors = connectors || plugin.getInfo().connectors || []
+  debug('quoting', amount, 'via', quoteConnectors)
```

Comment:
Should log the destination account and whether that amount is the source or destination amount